sensor/common/scan/scan.go

```diff
@@ -297,10 +299,23 @@ func (s *LocalScan) getPullSources(srcImage *storage.ContainerImage) []*storage.
 			continue
 		}
 
+		// This ID assignment addresses an edge case where a podspec references an image
+		// by tag (e.g. latest) and the registry contents for that tag change after the pod was
+		// created. We want to ensure that we pull metadata and layers based on the
+		// ID (digest) of the running image instead of what the tag currently represents.
+		// This condition will only be true for mirrors setup via the ImageTagMirrorSet (ITMS) CR,
+		// the other supported CRs, ImageContentSourcePolicy (ICSP) and ImageDigestMirrorSet (IDMS),
```

Comment:
nit: I got a bit confused when reading this. I think this may help, if I'm reading this right

```suggestion
		// This condition will only be true for mirrors setup via the ImageTagMirrorSet (ITMS) CR.
		// The other supported CRs, ImageContentSourcePolicy (ICSP) and ImageDigestMirrorSet (IDMS),
```